packages/flutter/lib/src/widgets/splash.dart

```diff
@@ -0,0 +1,621 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'lookup_boundary.dart';
+import 'notification_listener.dart';
+import 'ticker_provider.dart';
+
+// Examples can assume:
+// late Color color;
+// late BuildContext context;
+
+/// Signature for the callback used by [Splash]es to obtain the
+/// appropriate [Rect].
+typedef RectCallback = Rect Function();
+
+/// A splash of [color] shown in response to a user gesture.
+///
+/// The [confirm] and [cancel] methods alter the splash based on the
+/// user's behavior. For example, a tap event might trigger a splash
+/// that's confirmed when the corresponding [GestureDetector.onTapUp]
+/// event is seen and canceled when the pointer is dragged out of the
+/// reference box.
+///
+/// See also:
+///
+/// * [SplashBox], the widget that paints splashes.
+/// * [SplashFactory], a class that creates splashes using a specific
+///   function signature.
+/// * [InkResponse], which creates splashes using a [SplashFactory].
+abstract class Splash {
```

Comment:
I'm wondering if we should keep the "feature" in the name and call it `SplashFeature`. While I also prefer names to be shorter, I think a "splash feature" stands for the class of splash that will display, while the actual "splash" is the instantiated effect. (Just like the relationship between a widget and an element.)

Not a strong opinion though.

Reply:
I guess we can agree to disagree 😸

Reply:
But to expand on this, I'm not exactly sure what the difference is between "the class of splash that will display" and "the instantiated effect". It seems to me like both phrases are applicable here.

Reply:
Say that there's a button, and tapping the button triggers `SplashA`. Quickly tapping the button for multiple times will cause the button to have multiple _splash instances_ at the same time. But the button still only have one _splash feature_ or _splash class_.

Reply:
Ah that makes sense, thanks!

With this in mind, I do believe that the current names are an accurate representation of what you're describing.

A Material design button uses 1 instance of `SplashBox`, `SplashController`, and `SplashFactory` to generate multiple `Splash` instances (i.e. the visual effects on the screen).
So to try to match your analogy here:

> Tapping a button in rapid succession will cause the button to have multiple `Splash` instances at the same time, but the button still only has one `SplashFactory`.

Reply:
That makes sense as well. :)